### ISSUE TITLE
TypeError: Cannot read property 'emitter' of undefined

### ISSUE THREAD

**Issue by widdde** [AllenFang/react-bootstrap-table#1646]
When upgrading from react bootstrap table v 4.0.0 to v4.0.5 I get 
TypeError: Cannot read property of 'emitter' of undefined
at TableHeaderColumn.componentWillReceiveProps

Fails in:
`
if(nextprops.dataField !== this.props.dataField) {
var emitter = nextprops.filter.emitter || {} < -----
}`

In my table I use the following options:

`<BootstrapTable data={this.state.data} cellEdit={this.cellEditProp} remote={this.remote}
options={{noDataText: "no Data",
onCellEdit=this.onCellEdit.bind(this),
clearSearch: true,
onDeleteRow: this.onDeleteRow.bind(this,
insertModalHeader: this.createCustomModalHeader,
insertModalFooter: this.createModalFooter,
onAddRow: this.onAddRow.bind(this)}}

insertRow={true}
deleteRow={true}
selectRow={{
mode: 'radio',
bgcolor: 'pink',
beforeDeleteRow: this.beforeDeleteRow.bind(this)}}

version='4'
hover search striped>
<TableHeaderColumns..
< /BootstrapTable >
`

I don't get this error in earlier versions of bootstrap table.. 
One of my tables works, but another (the above) with more edit and add options doesn't work at all. 

EDIT: All works OK in v 4.0.3 as well, but not in v.4.0.5.

EDIT2: It fails when I have this simple <TableHeaderColumn:
`<TableHeaderColumn dataField='pattern' tdStyle={{whiteSpace: 'normal', fontWeight: 'lighter' }}>
Validation</TableHeaderColumn>
`
When I remove this tableHeaderColumn the table works fine. This is very strange. I have the exact same setup on other TableHeaderColumns, and they work perfectly..

When I change the order of my TableHeaderColumns it starts to work. How is this possible? 
If I move down this TableHeaderColumn to under another TableHeaderColumn everything works, and when I move it back the table throws TypeError as said above.

If I move the column under this column everything works..
`<TableHeaderColumn datafield='required' dataFormat={this.BooleanFormatter}
dataAlign='center' width='120' thStyle={{ 'fontWeight: 'lighter'}}
editable={{'type': 'checkbox', options: {values: 'true:false'}}}>Is Required</TableHeaderColumn>
`

**Comment by marcionicolau** [comment]
In the first render of data, everything is ok, when the data change, throws the error below.

My ```<TableHeaderColumn />``` is generated by 

```js
  apiCols = info =>
    _.map(info, (k, index) => (
      <TableHeaderColumn
        key={k}
        isKey={index === 0}
        dataField={k}
        width={this.widthSetting(index)}
        dataFormat={this.chooseFormatter(index)}
      >
        {k}
      </TableHeaderColumn>
    ));
```

I set replace=true in ```<BootstrapTable />``` but the error persist. This occurs in 4.0.0 to 4.0.5

 
```
TableHeaderColumn.js:86 Uncaught TypeError: Cannot read property 'emitter' of undefined
    at TableHeaderColumn.componentWillReceiveProps (TableHeaderColumn.js:86)
    at ReactCompositeComponent.js:608
    at measureLifeCyclePerf (ReactCompositeComponent.js:73)
    at ReactCompositeComponentWrapper.updateComponent (ReactCompositeComponent.js:607)
    at ReactCompositeComponentWrapper.receiveComponent (ReactCompositeComponent.js:544)
    at Object.receiveComponent (ReactReconciler.js:122)
    at Object.updateChildren (ReactChildReconciler.js:107)
    at ReactDOMComponent._reconcilerUpdateChildren (ReactMultiChild.js:204)
    at ReactDOMComponent._updateChildren (ReactMultiChild.js:308)
    at ReactDOMComponent.updateChildren (ReactMultiChild.js:295)
    at ReactDOMComponent._updateDOMChildren (ReactDOMComponent.js:944)
    at ReactDOMComponent.updateComponent (ReactDOMComponent.js:758)
    at ReactDOMComponent.receiveComponent (ReactDOMComponent.js:720)
    at Object.receiveComponent (ReactReconciler.js:122)
    at Object.updateChildren (ReactChildReconciler.js:107)
    at ReactDOMComponent._reconcilerUpdateChildren (ReactMultiChild.js:204)
    at ReactDOMComponent._updateChildren (ReactMultiChild.js:308)
    at ReactDOMComponent.updateChildren (ReactMultiChild.js:295)
    at ReactDOMComponent._updateDOMChildren (ReactDOMComponent.js:944)
    at ReactDOMComponent.updateComponent (ReactDOMComponent.js:758)
    at ReactDOMComponent.receiveComponent (ReactDOMComponent.js:720)
    at Object.receiveComponent (ReactReconciler.js:122)
    at Object.updateChildren (ReactChildReconciler.js:107)
    at ReactDOMComponent._reconcilerUpdateChildren (ReactMultiChild.js:204)
    at ReactDOMComponent._updateChildren (ReactMultiChild.js:308)
    at ReactDOMComponent.updateChildren (ReactMultiChild.js:295)
    at ReactDOMComponent._updateDOMChildren (ReactDOMComponent.js:944)
    at ReactDOMComponent.updateComponent (ReactDOMComponent.js:758)
    at ReactDOMComponent.receiveComponent (ReactDOMComponent.js:720)
    at Object.receiveComponent (ReactReconciler.js:122)
    at Object.updateChildren (ReactChildReconciler.js:107)
    at ReactDOMComponent._reconcilerUpdateChildren (ReactMultiChild.js:204)
    at ReactDOMComponent._updateChildren (ReactMultiChild.js:308)
    at ReactDOMComponent.updateChildren (ReactMultiChild.js:295)
    at ReactDOMComponent._updateDOMChildren (ReactDOMComponent.js:944)
    at ReactDOMComponent.updateComponent (ReactDOMComponent.js:758)
    at ReactDOMComponent.receiveComponent (ReactDOMComponent.js:720)
    at Object.receiveComponent (ReactReconciler.js:122)
    at ReactCompositeComponentWrapper._updateRenderedComponent (ReactCompositeComponent.js:751)
    at ReactCompositeComponentWrapper._performComponentUpdate (ReactCompositeComponent.js:721)
    at ReactCompositeComponentWrapper.updateComponent (ReactCompositeComponent.js:642)
    at ReactCompositeComponentWrapper.receiveComponent (ReactCompositeComponent.js:544)
    at Object.receiveComponent (ReactReconciler.js:122)
    at Object.updateChildren (ReactChildReconciler.js:107)
    at ReactDOMComponent._reconcilerUpdateChildren (ReactMultiChild.js:204)
    at ReactDOMComponent._updateChildren (ReactMultiChild.js:308)
    at ReactDOMComponent.updateChildren (ReactMultiChild.js:295)
    at ReactDOMComponent._updateDOMChildren (ReactDOMComponent.js:944)
    at ReactDOMComponent.updateComponent (ReactDOMComponent.js:758)
    at ReactDOMComponent.receiveComponent (ReactDOMComponent.js:720)
```

**Comment by marcionicolau** [comment]
I could, temporally fix, using ```fix={{}}```  in  ```<TableHeaderColumn>```

```
  apiCols = info =>
    _.map(info, (k, index) => (
      <TableHeaderColumn
        key={k}
        isKey={index === 0}
        dataField={k}
        filter={{}}
        width={this.widthSetting(index)}
        dataFormat={this.chooseFormatter(index)}
      >
        {k}
      </TableHeaderColumn>
    ));
```

**Comment by AllenFang** [comment]
hi all, is that this error only happen on second time render and also you have change the columns order(position) or some column being added or deleted?

**Comment by widdde** [comment]
@AllenFang Yes, that is correct. In my case the some columns has been added on the second render (basic table vs a more advanced table with more information)

**Comment by AllenFang** [comment]
Fixed on `v4.0.6`, let me know if the problem still remain, thanks
